static/js/settings.js

```diff
@@ -540,13 +540,30 @@ function _setup_page() {
                 loading.destroy_indicator($("#upload_avatar_spinner"));
                 var url = data.avatar_url + '&stamp=' + exports.avatar_stamp;
                 $("#user-settings-avatar").expectOne().attr("src", url);
+                $("#user_avatar_delete_button").show();
                 exports.avatar_stamp += 1;
             }
         });
 
     }
 
     avatar.build_user_avatar_widget(upload_avatar);
+    if (page_params.avatar_source === 'G') {
```

Comment:
I put it in settings.js because the upload_avatar function was placed in settings.js.

But I guess, `build_user_avatar_widget` is the more appropriate place.